File: scaffold/src/main/java/top/xuqingquan/web/x5/MiddlewareWebChromeBase.java
package top.xuqingquan.web.x5;

import com.tencent.smtt.sdk.WebChromeClient;

public class MiddlewareWebChromeBase extends WebChromeClientDelegate {

    private MiddlewareWebChromeBase mMiddlewareWebChromeBase;

    protected MiddlewareWebChromeBase(WebChromeClient webChromeClient) {
        super(webChromeClient);
    }

    protected MiddlewareWebChromeBase() {
        super(null);
    }

    @Override
    public final void setDelegate(WebChromeClient delegate) {
        super.setDelegate(delegate);
    }

    public final MiddlewareWebChromeBase enq(MiddlewareWebChromeBase middlewareWebChromeBase) {
        setDelegate(middlewareWebChromeBase);
        this.mMiddlewareWebChromeBase = middlewareWebChromeBase;
        return this.mMiddlewareWebChromeBase;
    }


    public final MiddlewareWebChromeBase next() {
        return this.mMiddlewareWebChromeBase;
    }

}

File: scaffold/src/main/java/top/xuqingquan/web/x5/MiddlewareWebChromeBase.kt
package top.xuqingquan.web.x5

import com.tencent.smtt.sdk.WebChromeClient

open class MiddlewareWebChromeBase(webChromeClient: WebChromeClient? = null) :
    WebChromeClientDelegate(webChromeClient) {

    private var mMiddlewareWebChromeBase: MiddlewareWebChromeBase? = null

    override var delegate: WebChromeClient?
        get() = super.delegate
        set(delegate) {
            super.delegate = delegate
        }

    fun enq(middlewareWebChromeBase: MiddlewareWebChromeBase): MiddlewareWebChromeBase? {
        super.delegate = middlewareWebChromeBase
        this.mMiddlewareWebChromeBase = middlewareWebChromeBase
        return this.mMiddlewareWebChromeBase
    }

    operator fun next(): MiddlewareWebChromeBase? {
        return this.mMiddlewareWebChromeBase
    }

}
